### PR TITLE
Update homepage

### PR DESCRIPTION
**Pull request by luisbettencourt**
## 🌟 What is the purpose of this PR?

This PR implements the new BP homepage design.

## 🔗 Related links

- [Replace font](https://app.asana.com/0/1203543021352032/1203658814085284/f) _(internal)_
- [Update Header](https://app.asana.com/0/1203543021352032/1203658814085281/f) _(internal)_
- [Zero section](https://app.asana.com/0/1203179076056186/1202787635216908/f) _(internal)_
- [Supported env and cases section](https://app.asana.com/0/1203543021352032/1203611715443315/f) _(internal)_
- [Update block carousel section](https://app.asana.com/0/1203543021352032/1203624039082637/f) _(internal)_
- [Update signup section](https://app.asana.com/0/1203543021352032/1203658814085293/f) _(internal)_

- [Design](https://www.figma.com/file/Fpmde3E9nJMb4VLkPTYgrM/blockprotocol.org?node-id=4628%3A31441&t=lf6MnjOWavm21nDK-1) _(internal)_

## 🚫 Blocked by

- [x] Add Colfax font assets (used an HTML link tag for testing so far);
- [x] Validate if the new font made any other page's layout weird;
- [x] Fix tests;
- [x] Compress images.

## 🔍 What does this change?

Besides homepage-specific styling changes:
- Changed the `Apercu` font for `Colfax` accross the website;
- Added a `linkProps` to `LinkButton` to pass props directly to the `BaseLink` MUI component;
- Updated typography variant font-sizes using [this tool](https://utopia.fyi/type/calculator?c=320,14,1.25,1140,16,1.25,6,2,&s=0.75|0.5|0.25,1.5|2|3|4|6,s-l&g=s,l,xl,12), to better match the designs;
- Added a theme override (same one as being used on HASH) for MUI tooltips;
- Set the default `letter-spacing` to 0;
- Set the default button theme to have `font-family: "Inter"; letter-spacing: 0`;
- Made some style fixes to other pages that have been affected by the font change (mainly wordpress page);
- Removed the name of the key in the title of the api key generation modal (requested by @nonparibus).  

## 📹 Demo

image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by nonparibus** [comment]
Will take a look tomorrow AM when deployments are up. I think tests might need adjusting before this can merge in any case

**Comment by nonparibus** [comment]
@luisbettencourt We'll want to pull the PNGs out of this, do a compression pass, and deliver them via Cloudflare Images before this merges, too.

File: apps/site/src/components/pages/home/composable-interfaces.tsx [hunk 1]
```diff
@@ -1,81 +1,442 @@
...
+            }}
+          />
+        </Box>
+      }
+      placement="top"
     >
+      <LinkButton
+        linkProps={{
+          // eslint-disable-next-line no-useless-computed-key
+          ["data-testid"]: `${label}-button`,
```
Comment (luisbettencourt) [inline apps/site/src/components/pages/home/composable-interfaces.tsx:52]: Is there a better way to do this?

File: apps/site/src/components/pages/home/composable-interfaces.tsx [hunk 2]
```diff
@@ -1,81 +1,460 @@
...
+          },
+        }}
+      >
+        {icon}
+      </LinkButton>
+    </Tooltip>
+  );
+};
+
+export const ComposableInterfaces = () => {
```
Comment (CiaranMn) [inline apps/site/src/components/pages/home/composable-interfaces.tsx:76]: The name of this component no longer makes sense – it's more `SupportedApplications` or something like that now

Comment (luisbettencourt) [reply apps/site/src/components/pages/home/composable-interfaces.tsx]: I was overcomplicating this, thank you. Fixed and changed the component name to your suggestion in 96380cfcfc0a99ee124ea742150902fe85128e61

**Review by CiaranMn — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]
I have 2x minor tweaks I suggest we do in follow-up. Will DM directly. Thanks, @luisbettencourt - this is looking great!

**Review by CiaranMn — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**